### PR TITLE
Create BinarySearchTree.js

### PR DESCRIPTION
**Pull request by khushal87**
Refer issue #593 
Added binary Search tree

### CODE REVIEW

File: Binary_Search_Trees/BinarySearchTree.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+tree.insert(10);
+tree.insert(9);
+tree.insert(7);
+tree.lookup(10);
+//JSON.stringify(traverse(tree.root));
+
+
+
+
+
```
Comment (jainaman224) [inline Binary_Search_Trees/BinarySearchTree.js]: Remove extra new lines.

File: Binary_Search_Trees/BinarySearchTree.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+              currentNode.right=newnode;
+              return this;
+            }
+          currentNode=currentNode.right;
+        }
+      }
+    }
+  }
+
+  lookup(value)     //the loopup function
```
Comment (jainaman224) [inline Binary_Search_Trees/BinarySearchTree.js]: Typo in word lookup.

**Comment by khushal87** [comment]
I think the file name is correct as it matches the other files in that folder

**Comment by jainaman224** [comment]
~BinarySearchTree~ -> Binary_Search_Tree

**Comment by khushal87** [comment]
Done @jainaman224

**Pull request merged**